tensorboard/data/server/run.rs

```diff
@@ -70,6 +71,16 @@ struct TimeSeries {
     rsv: StageReservoir<StageValue>,
 }
 
+/// A value staged in the reservoir.
+///
+/// This is kept as close as possible to the on-disk event representation, since every record in
+/// the stream is converted into this format.
+#[derive(Debug)]
+struct StageValue {
+    wall_time: WallTime,
+    payload: EventValue,
+}
+
 impl TimeSeries {
```

Comment:
I have (a) noted this, (b) not really found it confusing, and
(c) considered calling one of them `StageTimeSeries` anyway, since it
meshes better with `StageValue` and `StageReservoir`. So maybe I’ll do
that, since you’ve raised this.

> Also feels slightly odd that in the stage we index by tag and store
> DataClass as auxiliary to the TimeSeries, whereas in the commit we
> have separate fields for each data class and index by tag within them

I’ve tried to keep the stage as close as possible to the representation
on disk, and the commit as close as reasonably possible to the data
provider interface, with the committing operation mapping between them.
This makes it easiest to keep loading throughput as high as possible.
So the discrepancy feels pretty natural to me. Certainly open to making
adjustments, especially as we start making optimizations like the one
that you mention.

Reply:
Yep. Have a commit drafted locally for this, too.